src/utils/ERC20CosmosCoinAmountConversion.sol

```diff
@@ -0,0 +1,137 @@
+     */
+    function _convertERC20AmountToCosmosCoin(
+        address tokenAddress,
+        uint256 amount
+    )
+        internal
+        view
+        returns (uint64 convertedAmount, uint256 remainder)
+    {
+        // Note if we want to allow different cosmos decimals, we need to handle that here too
+        // Note that tokenAddress input validations are executed in the _getERC20TokenDecimals function
+        uint8 tokenDecimals = _getERC20TokenDecimals(tokenAddress);
+        // TODO write an ADR for this?
+        require(tokenDecimals >= 6, "ERC20 with less than 6 decimals are not supported");
+        // TODO Add custom error?
+        // Amount input validation
+        require(amount != 0, "Requested conversion for the 0 amount");
+        // Ensure the amount respects the token's decimals
+        // Handle the case where the input amount exceeds the token's precision
+        uint256 temp_convertedAmount;
+        uint256 factor;
+        // Case ERC20 decimals are bigger than cosmos decimals
+        if (tokenDecimals > DEFAULT_COSMOS_DECIMALS) {
+            factor = 10 ** (tokenDecimals - DEFAULT_COSMOS_DECIMALS);
+            temp_convertedAmount = amount / factor;
+            remainder = amount % factor;
+        } else if (tokenDecimals < DEFAULT_COSMOS_DECIMALS) {
+            // Keeping this code here until final team decision is made.
+            // Restricting the support for decimals >= 6 this part will never be executed.
+            // Case ERC20 decimals < DEFAULT_COSMOS_DECIMALS
+            // Note we need to amplify the decimals
+            factor = 10 ** (DEFAULT_COSMOS_DECIMALS - tokenDecimals);
+            temp_convertedAmount = amount * factor;
+            remainder = 0;
+        } else {
+            // Case ERC20 decimals == DEFAULT_COSMOS_DECIMALS --> Amount will fit the uint64
+            // Note that we need to handle the case < and == differently
+            temp_convertedAmount = amount;
+            remainder = 0;
+        }
+        // TODO Add custom error?
+        // ~uint64(0) is the max value supported by uint64
+        // Probably this check can be removed - cannot think a way to trigger this
+        require(temp_convertedAmount <= ~uint64(0), "Converted amount exceeds uint64 limits");
+        // At this point we should be sure that we are not loosing precision
+        convertedAmount = uint64(temp_convertedAmount);
+        return (convertedAmount, remainder);
+    }
+
+    // Convert Cosmos coin amount to ERC20 token amount
+    // Assuming that we support only ERC20.decimlas()>=6
+    /**
+     * @notice Convert the uint64 Cosmos coin amount into ERC20 token amount uint256
+     * @param tokenAddress The address of the token contract
+     * @param amount The amount to be converted
+     * @return convertedAmount The amount converted to uint256 supported by ERC20 tokens
+     */
+    function _convertCosmosCoinAmountToERC20(
```

Comment:
Do we even need this function? I was thinking we wouldn't even use any decimals going from Cosmos -> ERC20?

Reply:
I think the utility of this may come in when we need to convert the sdkAmount to the ERC20 and the ERC20 has > decimals than 6. In this case to properly represent the amount we need to amplify the value. However if we are not going to use any decimals from Cosmos, then I'll need to take it into account. Since this assume that we will use a sdkCoin amount representation as 1,000,000.